drivers/rtc/rtc_ll_stm32.c

```diff
@@ -247,22 +382,29 @@ static int rtc_stm32_set_time(const struct device *dev, const struct rtc_time *t
 static int rtc_stm32_get_time(const struct device *dev, struct rtc_time *timeptr)
 {
 	struct rtc_stm32_data *data = dev->data;
-
 	uint32_t rtc_date, rtc_time;
-
 #if HW_SUBSECOND_SUPPORT
 	const struct rtc_stm32_config *cfg = dev->config;
 	uint32_t rtc_subsecond;
-#endif
+#endif	/* HW_SUBSECOND_SUPPORT */
+	int err;
 
-	int err = k_mutex_lock(&data->lock, K_NO_WAIT);
+	if (timeptr == NULL) {
+		LOG_ERR("NULL rtc_time pointer");
+		return -EINVAL;
+	}
+
+	memset(timeptr, -1, sizeof(struct rtc_time));
+
+	err = k_mutex_lock(&data->lock, K_NO_WAIT);
```

Comment:
Justification ?
Likely requires a dedicated commit as well.

Reply:
> Justification ?

We cannot operate on a NULL pointer !?